File: 2016/2-january/bronze/notlast/notlast.cpp
#include <bits/stdc++.h>
using namespace std;
ofstream fout ("notlast.out");
ifstream fin ("notlast.in");
#define For(x) for(int i = 0; i < x; i++)
#define For2(x) for(int j = 0; j < x; j++)
#define Forv(vector) for(auto& i : vector)
int main(){
    int n;
    fin>>n;
    vector<string> cows= {"Bessie", "Elsie", "Daisy", "Gertie", "Annabelle", "Maggie", "Henrietta"};
    vector<pair<int,string> > ls;
    Forv(cows){
        pair<int,string> p(0,i);
        ls.push_back(p);
    }
    For(n){
        string s;
        int t;
        fin >> s>>t;
        if (s == cows[0])
            ls[0].first += t;
        else if (s == cows[1])
            ls[1].first+=t;
        else if (s == cows[2])
            ls[2].first+=t;
        else if (s == "Gertie")
            ls[3].first+=t;
        else if (s == "Annabelle")
            ls[4].first+=t;
        else if (s == "Maggie")
            ls[5].first+=t;
        else
            ls[6].first+=t;
    }
    sort(ls.begin(),ls.end());
    int minm = ls[0].first;
    int index = 0;
    while(ls.size()>0 and ls[index].first == minm){
        ls.erase(ls.begin());
    }
    if(1<ls.size() and ls[index].first == ls[index+1].first){
        fout<<"Tie\n";
    }
    else if(ls.size()==0){
        fout<<"Tie\n";
    }
    else{
        fout<<ls[0].second<<"\n";
    }
    
    
}

File: 2016/2-january/bronze/notlast/notlast.py
#Bessie, Elsie, Daisy, Gertie, Annabelle, Maggie, and Henrietta
cows = ["Bessie", "Elsie", "Daisy", "Gertie", "Annabelle", "Maggie", "Henrietta"]
fin = open("notlast.in", "r");
fout = open("notlast.out", "w")
ls = [[0,s] for s in cows]
n = int(fin.readline().strip())
for _ in range(n):
    x = fin.readline().strip().split()
    s = x[0]
    t = int(x[1])
    if s == cows[0]:
        ls[0][0] += t
    elif s == cows[1]:
        ls[1][0]+=t
    elif s == cows[2]:
        ls[2][0]+=t
    elif s == "Gertie":
        ls[3][0]+=t
    elif s == "Annabelle":
        ls[4][0]+=t
    elif s == "Maggie":
        ls[5][0]+=t
    elif s == "Henrietta":
        ls[6][0]+=t
ls.sort()
minm = ls[0][0]
while len(ls) > 0 and ls[0][0] == minm:
    ls.pop(0)
if len(ls) > 1 and ls[0][0] == ls[1][0]:
    fout.write("Tie\n")
elif len(ls) == 0:
    fout.write("Tie\n")
else:
    fout.write(ls[0][1]+"\n")
fout.close()